### PR TITLE
Get presentation example

### PR DESCRIPTION
**Pull request by johanneslundsten**

### CODE REVIEW

File: example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt [hunk 1]
```diff
@@ -145,6 +176,71 @@ class TruidSignupFlow(
...
+            tokenResponse.accessToken
+        } else {
+            //If access token is expired, use refresh token to get a new one
+            refreshToken()
+            getPersistedToken()!!.first.accessToken
+        }
+    }
+
+    private suspend fun refreshToken() {
+        // Check if token was refreshed by another thread, two refreshes with same refresh token
```
Comment (johanneslundsten) [inline example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt]: Not true in our case, but might be good to point out as we might want it in the future?

File: example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt [hunk 2]
```diff
@@ -145,6 +176,71 @@ class TruidSignupFlow(
...
+        // invalidates all access tokens and refresh tokens in accordance to
+        // https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-15#section-4.12.2
```
Comment (johanneslundsten) [inline example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt:221]: Not true in our case, but might be good to point out so we don't ruin customers consent if we want to introduce it?

Comment (johanneslundsten) [reply example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt]: Updated the sentence to 
```
        // Synchronized, two refreshes with same refresh token
        // invalidates all access tokens and refresh tokens in accordance to
        // https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-15#section-4.12.2
```

Comment (johanblumenberg) [reply example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt]: 👍

File: example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt [hunk 3]
```diff
@@ -96,6 +108,8 @@ class TruidSignupFlow(
         }
     }
 
+
+
```
[the remark this answers is not in the document]
Comment (johanneslundsten) [reply example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt]: Added ktlint now

File: example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt [hunk 4]
```diff
@@ -195,4 +294,16 @@ class TruidSignupFlow(
...
         return session.attributes["oauth2-code-verifier"] as String?
     }
+
+    private fun clearPersistence() {
+        _persistedTokenResponse = null
+    }
+
+    private fun persist(tokenResponse: TokenResponse) {
+        //Subtract 5 seconds on the real expiry to create a small buffer
+        _persistedTokenResponse = Pair(tokenResponse, LocalDateTime.now().plusSeconds(tokenResponse.expiresIn - 5))
```
[the remark this answers is not in the document]
Comment (johanneslundsten) [reply example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt]: Removed check of expiry and also this

File: example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt [hunk 5]
```diff
@@ -145,6 +177,73 @@ class TruidSignupFlow(
...
+            .header(AUTHORIZATION, "Bearer $accessToken")
+            .retrieve()
+            .awaitBody()
+    }
+
+    private suspend fun getAccessToken(): String {
+        val (tokenResponse, expires) = getPersistedToken() ?: throw Forbidden(
+            "access_denied",
+            "No access_token or refresh_token found"
+        )
```
[the remark this answers is not in the document]
Comment (johanneslundsten) [reply example-backend/src/main/kotlin/app/truid/example/examplebackend/TruidSignupFlow.kt]: Removed expiry check and always use refresh

**Review by johanblumenberg — APPROVED** [review]

**Review by marten-w — APPROVED** [review]

**Pull request merged**